.github/workflows/deploy_go_lambda_gf.yml

```
@@ -0,0 +1,66 @@
+name: utility-deploy-go-lambda (Greenfield)
+
+on:
+  workflow_call:
+    inputs:
+      env:
+        description: AWS environment to deploy to
+        required: true
+        type: string
+      project:
+        description: Project name
+        required: true
+        type: string
+      go_files:
+        description: space-delimited go files
+        required: true
+        type: string
+
+jobs:
+  deploy:
+    name: "Build and Deploy Lambda"
+    permissions:
+      contents: read
+      id-token: write
+    runs-on: codebuild-dpc-app-${{github.run_id}}-${{github.run_attempt}}
+    defaults:
+      run:
+        working-directory: ./lambda/${{ inputs.project }}
+    environment: ${{ inputs.env }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+      - name: Build ${{ inputs.project }} zip file
+        env:
+          CGO_ENABLED: 0
+        run: |
+          go build -o bootstrap ${{ inputs.go_files }}
+          zip function.zip bootstrap
+      - name: AWS Credentials (non-prod)
+        if: ${{ inputs.env == 'dev' || inputs.env == 'test' }}
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ vars.AWS_REGION }}
+          role-to-assume: arn:aws:iam::${{ secrets.NON_PROD_ACCOUNT_ID }}:role/delegatedadmin/developer/dpc-${{ inputs.env }}-github-actions
+      - name: AWS Credentials (prod)
+        if: ${{ inputs.env == 'prod-sbx' || inputs.env == 'prod' }}
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ vars.AWS_REGION }}
+          role-to-assume: arn:aws:iam::${{ secrets.PROD_ACCOUNT_ID }}:role/delegatedadmin/developer/dpc-${{ inputs.env }}-github-actions
+
+      - name: retrieve s3 bucket
+        id: retrieve_s3_bucket
+        run: |
+          S3_LOOKUP_KEY="/dpc/${{ inputs.env }}/dpc-${{ inputs.env }}-${{ inputs.project }}-bucket"
+          LAMBDA_BUCKET_NAME=$(aws ssm get-parameter --name "$S3_LOOKUP_KEY" --with-decryption --query "Parameter.Value" --output text)
+          echo "lambda_bucket_name=$LAMBDA_BUCKET_NAME" >> "$GITHUB_OUTPUT"
+
+      - name: Upload and reload
+        env:
+          LABEL: ${{ inputs.env }}-${{ inputs.project }}
+        run: |
+          aws s3 cp --no-progress function.zip \
+            s3://${{ steps.retrieve_s3_bucket.outputs.lambda_bucket_name }}/function.zip
+          aws lambda update-function-code --function-name dpc-$LABEL \
+            --s3-bucket ${{ steps.retrieve_s3_bucket.outputs.lambda_bucket_name }} --s3-key function.zip
```

Hey, this works! However, I know there's desire for standardizing on some platform-provided patterns like usage of [cmsgov/ab2d-bcda-dpc-platform/actions/aws-params-env-action](https://github.com/CMSgov/ab2d-bcda-dpc-platform/tree/main/actions/aws-params-env-action). It's little different, but... what do you think of this?
```suggestion
      - name: retrieve s3 bucket
        uses: cmsgov/ab2d-bcda-dpc-platform/actions/aws-params-env-action@main
        env:
          AWS_REGION: ${{ vars.AWS_REGION }}
        run: |
          LAMBDA_BUCKET_NAME=/dpc/${{ inputs.env }}/dpc-${{ inputs.env }}-${{ inputs.project }}-bucket

      - name: Upload and reload
        env:
          LABEL: ${{ inputs.env }}-${{ inputs.project }}
        run: |
          aws s3 cp --no-progress function.zip \
            "s3://${LAMBDA_BUCKET_NAME}/function.zip"
          aws lambda update-function-code --function-name "dpc-${LABEL}" \
            --s3-bucket "$LAMBDA_BUCKET_NAME" --s3-key function.zip
```

---

Thanks for flagging. I forgot about the platform params env action!

---

Just had to make one small tweak to use 
`params:` and it looks like it's working. Tested [HERE](https://github.com/CMSgov/dpc-app/actions/runs/15003766203/job/42157124459)